test_runner/cloud_regress/test_cloud_regress.py

```diff
@@ -0,0 +1,81 @@
+"""
+Run the regression tests on the cloud instance of Neon
+"""
+
+import os
+import re
+import subprocess
+
+import psycopg2
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import RemotePostgres
+
+
+@pytest.mark.timeout(7200)
+@pytest.mark.remote_cluster
+def test_cloud_regress(remote_pg: RemotePostgres):
+    """
+    Run the regression tests
+    """
+    cur_test = os.environ.get("PYTEST_CURRENT_TEST")
+    assert cur_test
+    pg_version_match = re.search(r"\-pg(\d+)\]", cur_test)
+    assert pg_version_match
+    pg_version = int(pg_version_match.group(1))
+    with psycopg2.connect(remote_pg.connstr()) as conn:
+        with conn.cursor() as cur:
+            cur = conn.cursor()
+            cur.execute("SELECT COUNT(*) FROM pg_extension WHERE extname = 'regress_so'")
+            for r in cur:
+                num_ext = r[0]
+            assert int(num_ext) < 2
+            if num_ext == 1:
+                log.info("The extension is found")
+            else:
+                log.info("Creating the extension")
+                cur.execute("CREATE EXTENSION regress_so")
+                conn.commit()
+
+            log.info("Creating a C function to check availability of regress.so")
+            cur.execute(
+                "CREATE FUNCTION get_columns_length(oid[]) "
+                "RETURNS int AS 'regress.so' LANGUAGE C STRICT STABLE PARALLEL SAFE;"
+            )
```

Comment:
This is also a workaround for the full path problem

Reply:
Should we fix the problem with the full path? We know the approach, right?

Reply:
If I understand correctly, this problem depends on the environment variables passing through the proxy problem.
PG_LIBDIR is not set due to this.

Reply:
The problem is that pg_regress test setup sets libdir part of the path explicitly
https://github.com/neondatabase/postgres/blob/REL_16_STABLE_neon/src/test/regress/sql/test_setup.sql#L10

instead of using `$libdir` macro as all other extensions do
https://github.com/neondatabase/postgres/blob/3ec6e2496f64c6fec35c67cb82efd6490a6a4738/src/pl/plpgsql/src/plpgsql.control#L4

We can either fix handling of this in our extension request code (this will help other extensions that use full path as well)
or  patch `/test_setup.sql` to make it use $libdir

```
diff --git a/src/test/regress/sql/test_setup.sql b/src/test/regress/sql/test_setup.sql
index 1b2d434683..20c45b903c 100644
--- a/src/test/regress/sql/test_setup.sql
+++ b/src/test/regress/sql/test_setup.sql
@@ -7,7 +7,7 @@
 \getenv libdir PG_LIBDIR
 \getenv dlsuffix PG_DLSUFFIX
 
-\set regresslib :libdir '/regress' :dlsuffix
+\set regresslib '$libdir/regress' :dlsuffix
 
 --
 -- synchronous_commit=off delays when hint bits may be set. Some plans change
```

Reply:
This is a good solution, but it's a workaround.
I believe https://github.com/neondatabase/postgres/pull/500 is a true solution for this problem.